copter/source/docs/high-loop-rate-tuning.rst

```diff
@@ -0,0 +1,41 @@
+
+.. note:: this is much more computationally intensive, resulting in higher CPU loads. On H7 based autopilots it is acceptable, but F4 based autopilots might not be able to sustain this increased load.
+
+To setup running the PID rate loops closer to the higher gyro sampling rates, rather than the :ref:`SCHED_LOOP_RATE<SCHED_LOOP_RATE>`:
+
+Configuration
+=============
+There are only two new settings for enabling fast rate attitude, but these have an effect on several other settings:
+
+- :ref:`FSTRATE_ENABLE<FSTRATE_ENABLE>`:
+    - 0: Disables this feature.
+    - 1: Enable dynamic fast rate. On high CPU loading (as determined by either the main loop being slow or the gyro sample queue filling up) the rate will be reduced to an integer divisor of the gyro rate until the CPU stabilizes. This is the safest way of trying the support.
+    - 2: Enable dynamic fast rate, but fix to the requested rate on arming. This allows un-armed operations such as log download to proceed effectively. Note that you must be sure that you have sufficient CPU to enable this option or control can be compromised.
+    - 3: Enable fixed fast rate. This is useful for testing but not usually used.
+
+- :ref:`FSTRATE_DIV<FSTRATE_DIV>`: this defaults to 1 and represents the maximum fast rate gyro divisor. So a value of 2 will yield a rate loop of gyro rate / 2 (:ref:`INS_GYRO_RATE<INS_GYRO_RATE>` / 2). This allows a fixed fast rate that is lower than the gyro rate. This parameter is can be modified in-flight.
+
+The easiest way to try this feature is to set :ref:`FSTRATE_ENABLE<FSTRATE_ENABLE>` = 1 (Dynamic rate). If your flight controller is struggling it will reduce the attitude loop rate until normality is restored. :ref:`FSTRATE_ENABLE<FSTRATE_ENABLE>` = 2 (Fixed Rate) is not recommended until you have seen reasonable performance with dynamic rates.
+
+One way to gain more CPU is to switch off additional IMUs using :ref:`INS_ENABLE_MASK<INS_ENABLE_MASK>` or reduce :ref:`INS_GYRO_RATE<INS_GYRO_RATE>`. The main CPU load is from the INS and attitude control threads.
+
+When this feature is enabled, several other settings are affected:
+
+- :ref:`SERVO_DSHOT_RATE<SERVO_DSHOT_RATE>` - this represents a multiplier of the main loop rate to output to the motors. When the fast rate loop is enabled this is normally set to 1, however if the calculated rate will not exceed the gyro rate, higher values are allowed. For instance if :ref:`INS_GYRO_RATE<INS_GYRO_RATE>` = 2(2KHz) and :ref:`FSTRATE_DIV<FSTRATE_DIV>` = 4 then values of :ref:`SERVO_DSHOT_RATE<SERVO_DSHOT_RATE>` up to 4(4x loop rate) are allowed.
```

Comment:
When using the fast rate this is a multiple of the *fast rate* rate rather than the main loop rate. 

So in your example:
INS_GYRO_RATE = 2 - is 4Khz (2^2)
FSTRATE_DIV = 4 - fast rate rate is 4Khz/4 = 1Khz
The it is true that
SERVO_DSHOT_RATE up to 4 is allowed